### DISCUSSION TITLE
Branching

### DISCUSSION THREAD

**Discussion by tundeojediran** [t1-loanwise/loan-wise-Group1-Team1A#2] [General]
@t1-loanwise/a-team @t1-loanwise/b-team, a few things to note when creating branches for the feature you are working on:

- create your branch from the `dev` branch.

- your branch name should describe the feature you are working on or what you are trying to achieve. For example, if I am working on the login page, here are some examples of branch names: `login`, `login-page` etc

**Comment by Temidayo24** [comment]
Good morning sir.

Are you saying once we create a branch that describes what we have worked on, we are to delete such a branch and create another?

**Reply by tundeojediran** [reply]
you don't have to delete it. 
Let me use an example:
Let's say I am working on the login feature. I created a branch `login` from `dev` When I am done with my work, I will create a PR back to `dev`. After my login PR has been approved, I merge my login PR to the `dev` branch.

What that means is your `dev` branch has your login work in it.

If the next feature I am working on is registration, I will now create a branch `registration` from `dev`

**Comment by MUFTIATBAKARE** [comment]
Morning Supervisor,

Ok. So, having created a muftiat_branch to work on some components, are you saying I should rename to what I did

Secondly, this means I can no longer work in the said branch for my current task(reset password page), right? So, I have to create a branch named (reset_, password)

**Reply by tundeojediran** [reply]
you can ignore renaming for`muftiat_branch`, But going forward, your branch name should describe what you are doing.

**Reply by MUFTIATBAKARE** [reply]
Thank you

**Comment by MUFTIATBAKARE** [comment]
Ok, duly noted.

**Comment by Temidayo24** [comment]
Alright sir, thank you. But wouldn't the number of branches we have too many? or it doesn't matter?

**Reply by tundeojediran** [reply]
No, it doesn't matter. Later on, when the branches are too much, we can do some cleanups
